### PR TITLE
Self PR for line commenting purposes

### PR DESCRIPTION
**Pull request by ktakagaki**

### CODE REVIEW

**Comment by ktakagaki** [comment]
@dlwh
I'd like to try to understand/learn/refactor the breeze backend a bit.
(Thinking of nD tensors and immutable)

It would be great if you find time to comment/enlighten me on the line comments I post here.
If it's OK, I'll tag you for my questions.

Thanks,
Kenta

File: math/src/main/scala/breeze/linalg/Tensor.scala [hunk 1]
```diff
@@ -66,15 +66,51 @@ trait QuasiTensor[@spec(Int) K, @spec(Double, Int, Float, Long) V] {
...
+    * @see [[activeIterator]]
+    */
   def iterator: Iterator[(K, V)]
+  /**
+    * Provides a index key -> value pair iterator for all active elements of a sparse [[Tensor]].
+    * For dense [[Tensor]] implementations, equivalent to [[iterator]].
+    *
+    * @see [[iterator]]
+    */
   def activeIterator: Iterator[(K, V)]
```
Comment (ktakagaki) [inline math/src/main/scala/breeze/linalg/Tensor.scala]: @dlwh
The above comments are correct, right?

File: math/src/main/scala/breeze/linalg/support/package.scala [hunk 2]
```diff
@@ -24,7 +24,19 @@ package breeze.linalg
...
+    * @tparam From Orininal
+    * @tparam A
+    * @tparam B
+    * @tparam To
+    */
   type CanMapValues[From, A, B, To] = mapValues.Impl2[From, A=>B, To]
+  /**
+    * @see CanMapValues
+    */
   type CanMapActiveValues[From, A, B, To] = mapActiveValues.Impl2[From, A=>B, To]
```
Comment (ktakagaki) [inline math/src/main/scala/breeze/linalg/support/package.scala]: @dwlh
Could it be that the naming of the above two types is confusing with ```trait CanTraverseValues[From, A]```?
Something like ```MapValuesImpl2``` ```MapActiveValuesImpl2```?

File: math/src/main/scala/breeze/linalg/support/CanTraverseValues.scala [hunk 3]
```diff
@@ -25,9 +25,12 @@ import breeze.math.Complex
...
  */
 trait CanTraverseValues[From, A] {
-  /**Traverses all values from the given collection. */
+
+  /** Traverses all values from the given collection. */
   def traverse(from: From, fn: ValuesVisitor[A]): Unit
-  def isTraversableAgain(from: From):Boolean
+
+  @deprecated
+  final def isTraversableAgain(from: From): Boolean = true
```
Comment (ktakagaki) [inline math/src/main/scala/breeze/linalg/support/CanTraverseValues.scala]: @dwlh
Can  ```isTraversableAgain``` be deprecated?
I may be overlooking something important, but I don't think this is ever false within breeze.
Was this meant for some type of Iterator-like Traversal, which was eliminated at some point?

File: math/src/main/scala/breeze/linalg/NumericOps.scala [hunk 4]
```diff
@@ -25,6 +25,8 @@ import scala.reflect.ClassTag
 
 
 trait ImmutableNumericOps[+This] extends Any {
+
+  //ToDo: Could this be renamed to be more informative? Kind of sounds like Python repr()
   def repr: This
```
Comment (ktakagaki) [inline math/src/main/scala/breeze/linalg/NumericOps.scala]: @dlwh
What does ```repr``` stand for exactly?

Comment (dlwh) [inline math/src/main/scala/breeze/linalg/NumericOps.scala]: representation. The scala collections library uses this too in TraversableLike, XLike, etc.

File: math/src/main/scala/breeze/linalg/Tensor.scala (hunk 1 above)
Comment (dlwh) [inline math/src/main/scala/breeze/linalg/Tensor.scala]: yep

File: math/src/main/scala/breeze/linalg/support/CanTraverseValues.scala (hunk 3 above)
Comment (dlwh) [inline math/src/main/scala/breeze/linalg/support/CanTraverseValues.scala]: i thought there was an Iterator implementation of CanTraverseValues? There should be.

File: math/src/main/scala/breeze/linalg/support/package.scala [hunk 5]
```diff
@@ -24,7 +24,19 @@ package breeze.linalg
  * @author dlwh
  **/
 package object support {
+
+  /**
+    *
+    *
+    * @tparam From Orininal
```
Comment (dlwh) [inline math/src/main/scala/breeze/linalg/support/package.scala]: Original?

File: math/src/main/scala/breeze/linalg/support/package.scala (hunk 2 above)
Comment (dlwh) [inline math/src/main/scala/breeze/linalg/support/package.scala]: There used to be traits CanMapActiveValues and CanMapValues. I UFuncified them and left the type aliases to not break old code.

Comment (dlwh) [inline math/src/main/scala/breeze/linalg/support/package.scala]: I plan to do the same thing to CanTraverseValues, fwiw

File: math/src/main/scala/breeze/linalg/support/CanTraverseValues.scala (hunk 3 above)
Comment (dlwh) [inline math/src/main/scala/breeze/linalg/support/CanTraverseValues.scala]: there is. There's an implementation for TraversableOnce

File: math/src/main/scala/breeze/linalg/support/package.scala (hunk 2 above)
Comment (ktakagaki) [inline math/src/main/scala/breeze/linalg/support/package.scala]: I see, makes sense

File: math/src/main/scala/breeze/linalg/support/CanTraverseValues.scala (hunk 3 above)
Comment (ktakagaki) [inline math/src/main/scala/breeze/linalg/support/CanTraverseValues.scala]: I must be making real newbie mistake... where exactly is this implementation, I can't seem to find it in the master branch? Sorry

Comment (dlwh) [inline math/src/main/scala/breeze/linalg/support/CanTraverseValues.scala]: https://github.com/scalanlp/breeze/blob/f4c326f9f219859156a8c55bc667369813fa4b52/math/src/main/scala/breeze/linalg/support/CanTraverseValues.scala#L104

**Pull request closed without merging**